### PR TITLE
Historical classes

### LINKED ISSUE #214: Can't drag 6.041A from the audit (only on courseroad.mit.edu)
Not sure if a fresh push to prod will fix this, but I am unable to drag 6.041 in the 6-2 requirements list on to my road, but other subjects near it like 6.008 drag fine.  Weirdly, not seeing this issue on localhost.

### PR DESCRIPTION
**Pull request by georgiashay**
Closes #214 

Adds warning if a class is on the road after the last semester it was offered, and assumes the class is OK if it is on the road before the last semester it was offered.

Adds warning in the class info card for all classes that are no longer offered.

### CODE REVIEW

**Comment by georgiashay** [comment]
Hmm, that sounds like a good idea but I'm not sure my opinion.  We should definitely indicate that it's probably not going to be offered, but then all the future semesters would be yellow so there wouldn't be any difference between Fall/Spring if it used to be offered in one or the other.  This might be okay though, since we can't really say anything about what semester it would be offered if it's historical?

**Comment by npfoss** [comment]
Yeah I agree. Decided in person we should make all the future semesters yellow

**Comment by georgiashay** [comment]
Changed color of semester.  I also added a different message for when the class is no longer offered, but I can't see it due to #259.

File: src/components/Semester.vue [hunk 1]
```diff
@@ -191,10 +209,25 @@ export default {
     isSameYear: function () {
       return Math.floor((this.index - 1) / 3) === Math.floor((this.currentSemester - 1) / 3);
     },
+    noLongerOffered: function() {
+      if(this.itemAdding.is_historical) {
+        let lastSemester = this.itemAdding.source_semester.split("-");
+        let sourceSemester = ["fall", "IAP", "spring"].indexOf(lastSemester[0]);
+        //which class year the last year offered corresponds to; +1 if fall because fall semester year is off by 1
+        let sourceYear = parseInt(lastSemester[1]) - this.baseYear + (sourceSemester === 0 ? 1 : 0);
+        let lastSemesterNumber = sourceYear * 3 + sourceSemester + 1;
```
[the remark this answers is not in the document]
Comment (georgiashay) [reply src/components/Semester.vue]: I had been using 6.041A/B but I think there were plenty of others as well

File: src/components/Semester.vue [hunk 2]
```diff
@@ -194,10 +212,24 @@ export default {
...
+        const sourceSemester = ['fall', 'IAP', 'spring'].indexOf(lastSemester[0]);
+        // which class year the last year offered corresponds to; +1 if fall because fall semester year is off by 1
+        const sourceYear = parseInt(lastSemester[1]) - this.baseYear + (sourceSemester === 0 ? 1 : 0);
+        const lastSemesterNumber = sourceYear * 3 + sourceSemester + 1;
+        if (this.index > lastSemesterNumber) {
+          return true;
+        }
+      }
+      return false;
+    },
```
Comment (npfoss) [inline src/components/Semester.vue]: I feel like this `noLongerOffered` function could be used up above where the exact same code is copied for adding the warning (with some small modification to say what year, etc.)

**Pull request merged**